transform_hoist_assign.ml

```diff
@@ -13,12 +13,13 @@ open Instr
  * We only look at our own use-def chain. Thus the transformation renames the
  * variable to avoid overriding unrelated uses of the same name.
  *)
-let hoist_assignment (instrs : instruction_stream) : instruction_stream option =
-  let reaching = Analysis.reaching instrs in
-  let uses = Analysis.used instrs in
-  let dominates = Analysis.dominates instrs in
+let hoist_assignment ((formals, instrs) as inp : analysis_input) : analysis_input option =
+  let open Analysis in
+  let reaching = reaching inp in
+  let uses = used inp in
```

Comment:
the name used on both sign of the equal sign is different here, and the same on the two other cases. Which of `uses` or `used` is better?